File: creating-non-existing-dates/ruby/invalid_dates.rb
require 'date'

# Date.new with valid input
date = Date.new(2021, 1, 15)

puts date # prints 2021-01-15

# Date.new with invalid input
begin
  Date.new(2021, 2, 30)
rescue => e
  puts e # prints invalid date
end

File: creating-non-existing-dates/javascript/invalidDates.js
// new Date with valid input
const validDate = new Date(2021, 0, 15);

console.log(validDate.toDateString()); // prints "Fri Jan 15 2021"

// new Date with invalid input
const invalidDate = new Date(2021, 1, 30);

console.log(invalidDate.toDateString()); // prints "Tue Mar 02 2021"
